kube/src/api/admission.rs

```diff
@@ -0,0 +1,273 @@
+    /// requested. e.g. a patch can result in either a CREATE or UPDATE
+    /// Operation.
+    pub operation: Option<Operation>,
+    /// Information about the requesting user.
+    pub user_info: Option<UserInfo>,
+    /// The object from the incoming request.
+    pub object: Option<T>,
+    ///  The existing object. Only populated for DELETE and UPDATE requests.
+    pub old_object: Option<T>,
+    /// Specifies that modifications will definitely not be persisted for this
+    /// request.
+    #[serde(default)]
+    pub dry_run: bool,
+    /// The operation option structure of the operation being performed. e.g.
+    /// `meta.k8s.io/v1.DeleteOptions` or `meta.k8s.io/v1.CreateOptions`. This
+    /// may be different than the options the caller provided. e.g. for a patch
+    /// request the performed [`Operation`] might be a [`Operation::CREATE`], in
+    /// which case the Options will a `meta.k8s.io/v1.CreateOptions` even though
+    /// the caller provided `meta.k8s.io/v1.PatchOptions`.
+    pub options: Option<RawExtension>,
+}
+
+/// The operation specified in an [`AdmissionRequest`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub enum Operation {
+    /// An operation that creates a resource.
+    CREATE,
+    /// An operation that updates a resource.
+    UPDATE,
+    /// An operation that deletes a resource.
+    DELETE,
+    /// An operation that connects to a resource.
+    CONNECT,
+}
+
+/// An outgoing [`AdmissionReview`] response.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct AdmissionResponse {
+    /// Identifier for the individual request/response. This must be copied over
+    /// from the corresponding AdmissionRequest.
+    pub uid: String,
+    /// Indicates whether or not the admission request was permitted.
+    pub allowed: bool,
+    /// Extra details into why an admission request was denied. This field IS NOT
+    /// consulted in any way if "Allowed" is "true".
+    pub result: Status,
+    /// The patch body. Currently we only support "JSONPatch" which implements
+    /// RFC 6902.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub patch: Option<Vec<u8>>,
+    /// The type of Patch. Currently we only allow "JSONPatch".
+    #[serde(skip_serializing_if = "Option::is_none")]
+    patch_type: Option<PatchType>,
+    /// An unstructured key value map set by remote admission controller (e.g.
+    /// error=image-blacklisted). MutatingAdmissionWebhook and
+    /// ValidatingAdmissionWebhook admission controller will prefix the keys with
+    /// admission webhook name (e.g.
+    /// imagepolicy.example.com/error=image-blacklisted). AuditAnnotations will
+    /// be provided by the admission webhook to add additional context to the
+    /// audit log for this request.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub audit_annotations: Option<HashMap<String, String>>,
```

Comment:
This can be `HashMap<String, String>`, with #[serde(skip_serializing_if = "HashMap::is_empty")]`.